pkg/server/invoke.go

```diff
@@ -31,13 +32,26 @@ func (s *DriverServiceServer) Invoke(ctx context.Context, req *pb.InvokeRequest)
 	s.log.Debug("Invoke request received", zap.Any("instance", req.Instance.Uuid), zap.Any("action", req.Method))
 	sp := req.GetServicesProvider()
 	client, err := one.NewClientFromSP(sp, s.log)
+	instance := req.GetInstance()
 	if err != nil {
 		return nil, status.Errorf(codes.InvalidArgument, "Error making client: %v", err)
 	}
 
-	action, ok := actions.Actions[req.GetMethod()]
+	method := req.GetMethod()
+
+	if _, ok := actions.AdminActions[method]; ok && (instance.GetAccessLevel() < access.SUDO) {
+		return nil, status.Errorf(codes.PermissionDenied, "Action %s is admin action", method)
+	}
+
+	// Machines are suspended when service is unpaid and shouldn't be available
+	if instance.GetStatus() == ipb.InstanceStatus_SUS && method != "resume" {
```

Comment:
this check must be rejecting ALL actions, when instance is suspended, and must be performed [here](https://github.com/slntopp/nocloud/blob/01aa56a049b2740cdf743db07ad48b91988f08c4/pkg/instances/server.go#L88-L101) instead of the driver